broker-core/src/main/java/io/zeebe/broker/subscription/message/state/SubscriptionState.java

```diff
@@ -30,7 +30,7 @@
 public class SubscriptionState<T extends Subscription> {
   private static final byte[] EXISTENCE = new byte[] {1};
   private static final int TIME_OFFSET = 0;
-  private static final int KEY_OFFSET = TIME_OFFSET + Long.BYTES;
+  public static final int KEY_OFFSET = TIME_OFFSET + Long.BYTES;
```

Comment:
The encoding of the key (i.e. tuple `(TIMESTAMP, KEY)`) is currently spread over three classes: `SubscriptionState`, `Subscription` implementations and `WorkflowState#findSubscription`. This is not so easy to understand, for example it was not immediately obvious to me that `Subscription` implementations must write the key in a certain format.

I'd prefer if this concern was encapsulated in `SubscriptionState` and I think we can achieve that by splitting the method `Subscription#writeKey` into `Subscription#writeTimestamp` and `Subscription#writeKey` (remainder).

Reply:
Funny in planning of doing the abstraction I wanted to do this, but somehow I forgot it :man_shrugging: 
Yes makes total sense will do.